### PR TITLE
Friendship 2

### PR DESCRIPTION
**Pull request by kelibst**
- Try to implement friendships with 2 rows for mutual friendship. Remember about unit and integrations tests!

### CODE REVIEW

File: app/controllers/friendship_controller.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      friend_obj = Friendship.new(user_id: current_user.id, friend_id: params[:id], confirmed: false)
+      friend_obj.save
+      flash[:success] = 'Friend request sent'
+    else
+      friend_obj = Friendship.where(user_id: current_user.id, friend_id: params[:id]).select('id')
+      Friendship.destroy(friend_obj.ids)
+      flash[:success] = 'Friend request cancelled'
+    end
+    redirect_to users_path
+  end
```
Comment (wathigo) [inline app/controllers/friendship_controller.rb:23]: This method is quit long. Can we split it into several methods, each having a single responsibility. Like we can have `create` method for only creating a friendship record, `destroy` for cancelling a friend request, `update` for accepting a friend request.
You can read more about single responsibility principle [here](https://www.sitepoint.com/solid-ruby-single-responsibility-principle/)

**Review by wathigo — CHANGES_REQUESTED** [review]
## CHANGES REQUIRED 
Great effort on this project @kelibst 
Let's make some changes in order to make the friendship feature more efficient.

### REQUIRED
* When friendship is confirmed, an additional row with reversed user-friend pair is created.
After accepting a friendship only one row was created as shown :point_down: 
Screenshot from 2020-05-30 09-55-23
* The database configurations are not correct. Let's remove user specific configs like username and password.
* We have some failing tests.
Screenshot from 2020-05-30 10-00-46
* See the comments for more pointer :point_down: 

**After implementing the changes, submit another request for review.**

**Good luck in your next submission :four_leaf_clover: 
Simon Wathigo**

**Comment by uryelah** [comment]
Hello **Keli Dogbevi**, I'm Sarah your last reviewer for this PR 👩‍🚀 

Great job, you did  some hardcore refactoring on the req_init method!
The inverse friendships look to be working too.

-------

## **Your project is approved! 🎊 🥅 💛**

**You may close this Pull Request and merge the changes**


---------

    
    
**Keep being an awesome dev and good luck with your next project!**

---------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Comment by kelibst** [comment]
Thanks Sarah

> Hello **Keli Dogbevi**, I'm Sarah your last reviewer for this PR 
> 
> Great job, you did some hardcore refactoring on the req_init method!
> The inverse friendships look to be working too.
> 
> ## **Your project is approved! confetti_ball goal_net yellow_heart**
> **You may close this Pull Request and merge the changes**
> 
> **Keep being an awesome dev and good luck with your next project!**
> 
> Cheers and Happy coding!
> 
> Code review by **Sarah Uryelah Silva Chamorro**
> 
> (Feel free to contact me on slack by the handle of **sarah** if you have any additional question)

**Pull request merged**